instagram-app/src/App.js

```diff
@@ -8,24 +8,66 @@ class App extends React.Component {
   constructor() {
     super();
     this.state = {
+      data: [],
+      searchText: ''
+    };
+  }
+
+  componentDidMount() {
+    console.log('CDM');
+    this.setState({
       data: dummyData
-    }
+    })
+  }
+
+ 
+  search = e => {
+    //filter the data array according the input of the seach bar. If the input matches, display the posts with that username
+    e.preventDefault();
+    let filteredData = [];
+    let currentData = this.state.data;
+
+    filteredData = currentData.filter(item => {
+      const lc = item.username.toLowerCase();
+      const filter = this.state.searchText.toLowerCase();
+
+      return lc.includes(filter);
+    })
+
+    this.setState({
+      data: filteredData
+    });
+
+
+
+  }
+
+  handleChanges = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
   }
+  
 
   render() {
     return (
       <div className="App">
-        <SearchBar /> 
+        <SearchBar 
+        handleChanges = {this.handleChanges}
+        value = {this.state.searchText}
+        search = {this.search}
+        /> 
         {
           this.state.data.map(item => {
```

Comment:
Here is where you need to apply the ternary in order to display the filtered data for your search function. Check my code snippet again - you missed part of it!